### PR TITLE
add DNS infra state and www redirect

### LINKED ISSUE #4165: Redirect www.wellcomecollection.org to wellcomecollection.org
Right now, if you visit <https://www.wellcomecollection.org/>, you get a Code Enigma error page:

Screenshot 2020-01-14 at 16 26 31

There's an A record in Route53 pointing a Code Enigma server: https://console.aws.amazon.com/route53/home?#resource-record-sets:Z3THRVQ5VDYDMC

We should:

- [x] Check with Kate what this server is doing, and if it's anything important
- [ ] If not, remove this A record, and replace it with a redirect to wellcomecollection.org

### PR DESCRIPTION
**Pull request by jamesgorrie**
### What is this PR trying to achieve?
Fixes #4165 

### Who is this change for?
People who type in `www.` and expecting a working website.

--- 

I've added this as part of a PR I meant to pull in ages ago.

### CODE REVIEW

File: infrastructure/terraform/dns/main.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
+locals {
+  weco_hosted_zone_id = "Z3THRVQ5VDYDMC"
+}
```
Comment (alexwlchan) [inline infrastructure/terraform/dns/main.tf]: You could also access this with an `aws_route53_zone` data source: https://www.terraform.io/docs/providers/aws/d/route53_zone.html

That’s what we do in the Archivematica stack: https://github.com/wellcomecollection/archivematica-infra/blob/33563750240d667fb6e0ea349919be8dc08b44ae/terraform/infra/certs.tf#L19-L23

Comment (jamesgorrie) [reply infrastructure/terraform/dns/main.tf]: Nice - thank you!

File: infrastructure/terraform/dns/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from here.
+
+## Purpose
+
+- To manage domains pointing to third-party services held outside of our
+  infrastructure.
+- Handling redirects.
+- To expose state data from the Routemaster account so other Terraform
+  provisionings can import that state. i.e. importing the Zone ID as
+  from the output `weco_hosted_zone_id`.
```
Comment (alexwlchan) [inline infrastructure/terraform/dns/README.md]: Is it worth including some copy/paste Terraform you can use to get that? i.e. include the explicit `terraform and locals` blocks.

File: infrastructure/terraform/dns/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+```sh
+1 error occurred:
+        * aws_route53_record.docs: 1 error occurred:
+        * aws_route53_record.docs: AccessDenied: User: arn:aws:sts::250790015188:assumed-role/wellcomecollection-assume_role_hosted_zone_update/1579081975684193000 is not authorized to perform: route53:GetChange on resource: arn:aws:route53:::change/C8Q2UIJJMKTI1
+        status code: 403, request id: c0498db6-7240-4072-a03c-466c7878fea1
+```
+
+This happens on the first apply, and won't happy again.
+
+We are looking into how to resolve it.
```
Comment (alexwlchan) [inline infrastructure/terraform/dns/README.md]: Should there be a GitHub issue for this as well?

Comment (jamesgorrie) [reply infrastructure/terraform/dns/README.md]: Wish there was one - it's living in D&T land somewhere.

File: infrastructure/terraform/dns/README.md (hunk 2 above)
Comment (jamesgorrie) [reply infrastructure/terraform/dns/README.md]: I might extract this out into it's own module to make it easier, trying to make the examples made me realise this is a bit too fiddly. Going to merge as it's actually applied though.

**Pull request merged**